pkg/resource/graph/dependency_graph.go

```diff
@@ -82,6 +82,79 @@ func (dg *DependencyGraph) DependingOn(res *resource.State,
 	return dependents
 }
 
+// DependingOnUniquely returns a slice containing all resources that directly or indirectly
+// depend upon *only* the given resource.  Resources that also depend on another resource with
+// the same URN will not be included in the returned slice.  The returned slice is guaranteed
+// to be in topological order with respect to the snapshot dependency graph.
+//
+// The time complexity of DependingOnUniquely is linear with respect to the number of resources.
+func (dg *DependencyGraph) DependingOnUniquely(res *resource.State) []*resource.State {
```

Comment:
This doesn't seem to do what it says. There's no URN uniqueness checks which should be needed to make this right. We should probably have unit tests for this function itself as well, not just the state edit function.

Reply:
Yeah, I was struggling to name this tbh.  Maybe `OnlyDependingOn` would be a better name?  Or do you have any other suggestions?  I do think the comment is correct though. 

I can add some unit tests.